### PR TITLE
Allow setting env for server

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: lib/server-starter.js [hunk 1]
```diff
@@ -40,15 +40,18 @@ export default class ServerStarter extends EventEmitter {
...
    * @param {object} [options] - Optional settings.
+   * @param {object} [options.env=process.env] - Environment key-value pairs.
    * @param {boolean} [options.stdout=false] - Forward server output from STDOUT to STDOUT.
    * @param {boolean} [options.stderr=true] - Forward server output from STDERR to STDERR.
    */
   launch(cmd, args, options = {}) {
     if (typeof this._process !== 'undefined') throw new Error('Server already launched');
     const stdout = typeof options.stdout !== 'undefined' ? options.stdout : false;
     const stderr = typeof options.stderr !== 'undefined' ? options.stderr : true;
+    const spawn_options = {stdio: ['pipe', 'pipe', 'pipe', this._fd]};
```
Comment (kraih) [inline lib/server-starter.js]: No snake case.

File: lib/server-starter.js [hunk 2]
```diff
@@ -40,15 +40,18 @@ export default class ServerStarter extends EventEmitter {
...
+    if (typeof options.env !== 'undefined') spawn_options.env = options.env;
```
Comment (kraih) [inline lib/server-starter.js]: `if (options.env !== undefined)` seems more efficient.

**Comment by kraih** [comment]
Once it works i'm in favor of the feature. 😁

File: lib/server-starter.js (hunk 2 above)
Comment (oschwald) [reply lib/server-starter.js]: I copied the pattern from the above instances. Should I update those as well?

**Comment by oschwald** [comment]
Odd, the tests do pass locally. I'll need to dig into what is going on in the CI.

File: lib/server-starter.js (hunk 2 above)
Comment (kraih) [reply lib/server-starter.js]: That would be nice. Some of this code needs a bit of cleanup. 🤔

**Comment by oschwald** [comment]
I had to pass through the `process.env` to make the tests pass within the GitHub Action environment. I assume it is related to how `node` is set up there. It should work now.

**Comment by kraih** [comment]
Please rebase on main and squash.

File: test/start_fd.js [hunk 3]
```diff
@@ -78,3 +78,18 @@ function getPort() {
     });
   });
 }
+
+t.test('Set env', async t => {
+  const server = await ServerStarter.newServer();
+  t.equal(server.pid, null);
+  await server.launch('node', ['test/support/server_fd.js'], {env: {...process.env, test: 'value'}});
+  t.equal(typeof server.pid, 'number');
+  const ua = new UserAgent({baseUrl: server.url()});
```
Comment (kraih) [inline test/start_fd.js]: This should be `baseURL`.

**Comment by oschwald** [comment]
I rebased against main.

**Comment by oschwald** [comment]
Thanks for merging!

**Comment by kraih** [comment]
Thanks for the patch!

**Pull request merged**